src/sorts/hybrid/DropMergeSort.java

```diff
@@ -49,18 +51,17 @@ public DropMergeSort(ArrayVisualizer arrayVisualizer) {
     private final int EARLY_OUT_TEST_AT = 4;
     private final double EARLY_OUT_DISORDER_FRACTION = 0.6;
     
-    private void truncateArrayList(ArrayList<Integer> arrayList, int len) {
+    private void truncateArrayList(List<Integer> arrayList, int len) {
     	int size = arrayList.size();
     	arrayList.subList(len, size).clear();
-        Writes.changeAllocAmount(len - size);
     }
     
     @Override
     public void runSort(int[] array, int length, int bucketCount) {
         if (length < 2) return;
         
         PDQBranchedSort pdqSort = new PDQBranchedSort(arrayVisualizer);
-        ArrayList<Integer> dropped = new ArrayList<>(length);
+        List<Integer> dropped = new ArrayVList(length);
```

Comment:
why not use the `ArrayVList` type, since that's the one you should always be using anyway?

Reply:
Because it's proper to use the interface name instead of the class name in Java (it makes it easier to read, as the reader doesn't have to know what type is used internally, and it also means that you don't need to change lots of code when changing the type used internally.)

Reply:
but the other sort uses `ArrayVList` because of the special method arguments (so consistency and anyway, using it emphasizes that this is something we are measuring)

Reply:
But it doesn't have to. It *could* use `Writes.arrayListAdd`.

Reply:
but as-is it's not (and in that case, you don't need a custom class, I think?)

Reply:
You need a custom class for visualization, because `toArray` is too slow for visualization, besides triggering a `ConcurrentModificationException`.

Reply:
I don't really know the internals of this stuff, what does `ArrayVList` do to allow for visualization, compared to `ArrayList`?

Reply:
It registers its internal array with ArrayVisualizer.arrays

Reply:
but why not just re-use how regular arrays work with `Reads` and `Writes`?

Reply:
What do you mean?

Reply:
with regular arrays, you use those proxy methods to interact with it instead of directly accessing it

Reply:
Maybe I should add `List` support to `Writes.write`. 
But that isn't directly related to ArrayVList.

Reply:
what does ArrayVList do that that doesn't do, then?

Reply:
It visualizes the array!

Reply:
but that's what the proxy methods do too

Reply:
Without ArrayVList, there's no performant way to visualize `List`s.

Reply:
yeah, probably should've clarified that my original question was what `ArrayVList` does to be performant

Reply:
ArrayVLists register themselves directly with the visualizer. The only way to visualize ArrayLists normally is by using `toArray` to *copy* the entire list every frame. `toArray` is also unviable because it triggers a `ConcurrentModificationException` 30% of the time.

Reply:
don't know about the internals, why can't you just directly read the ArrayList?

Reply:
To do that, you need to either use reflection to read private fields, or use `get`, which triggers a `ConcurrentModificationException`, just like `toArray`.

Reply:
why do regular arrays not have this issue?

Reply:
Because Java

Reply:
can't fight those facts